app/main.development.js

```diff
@@ -481,15 +496,21 @@ const launchDCRWallet = () => {
     }
   });
 
-  if (debug) {
-    dcrwallet.stdout.on("data", (data) => {
-      process.stdout.write(`${data}`);
-    });
+  dcrwallet.stdout.on("data", (data) => {
+    while (dcrwalletLogs.length + Buffer.from(data).length > MAX_LOG_LENGTH) {
+      dcrwalletLogs = dcrwalletLogs.slice(dcrwalletLogs.indexOf(dcrwalletLogs.slice(dcrwalletLogs.length-1, dcrwalletLogs.length),1)+1);
```

Comment:
This seems a bit more complicated and slower than it needs to be. I think this can be improved in the folowing ways:

1. Code for handling `dcrwallet.stdout.on(data)` and `dcrwallet.stderr.on(data)` is very similar. When it was 1 line for each it was ok to be duplicated, but since this is now more elaborate, this could be refactored into a local parametrized closure to unify the code. Bonus points for doing it once for all daemons. Something like:

```
const AddToLog = (destIO, destLogBuffer, data) => {
  ...  
  debug && destIO.write(data)
  return Buffer.concat(destLogBuffer, ...);
};
...
dcrwallet.stdout.on("data", (data) => dcrWalletLogs = AddToLog(process.stdout, dcrWalletLogs, data))
dcrwallet.stderr.on("data", (data) => dcrWalletLogs = AddToLog(process.stderr, dcrWalletLogs, data))
```

2. Don't call `buffer.from(data)` repeatedly. Store it on a local var then use that.

3. Use `os.EOL` instead of the slice(lenght-1, length) as it makes instantly more clear what is happening here :wink: 

4. Instead of a loop to figure out how many lines to remove from the top, search for the first EOL after the amount of bytes that need to be removed (data.length) and slice from there until the end. You can then drop the loop.

5. Current impl has an infinite bug when data.length > MAX_LOG_LENGTH. Tested by reducing MAX_LOG_LENGTH to 1000 (or other small value), when closing the app it gets stuck in an infinite loop.